src/ui_manager.py

```diff
@@ -290,21 +292,27 @@ def select_screen_object_match(match: TemplateMatch, delay_factor: tuple[float,
     mouse.click("left")
     wait(0.05, 0.09)
 
-def wait_for_screen_object(screen_object: ScreenObject, time_out: int = None) -> TemplateMatch:
+def wait_for_screen_object(screen_object: ScreenObject, time_out: float = 30) -> TemplateMatch:
     roi = Config().ui_roi[screen_object.roi] if screen_object.roi else None
-    time_out = time_out if time_out else 30
     match = TemplateFinder().search_and_wait(
         ref = screen_object.ref,
         time_out = time_out,
         threshold = screen_object.threshold,
         roi = roi,
-        best_match = screen_object.best_match,
         use_grayscale = screen_object.use_grayscale,
-        normalize_monitor = screen_object.normalize_monitor)
-    if match.valid:
-        return match
+        normalize_monitor = screen_object.normalize_monitor,
+        suppress_debug = screen_object.suppress_debug
+        )
     return match
 
+def wait_for_expiration(screen_object: ScreenObject, time_out: float = 3) -> bool:
```

Comment:
Not a big fan of the name here. Also makes sense to extract functions here. 

wait_for_visible = wait_for(lambda: is_visible, timeout=)

wait_for_hidden = wait_for(lambda: !is_visible, timeout=)